Add `r-twosamplemr` and missing dependencies

mfansler opened the pull request:
Adds [R-universe package `TwoSampleMR`](https://mrcieu.r-universe.dev/TwoSampleMR) as `r-twosamplemr` and the following required dependencies:

- `r-mr.raps` ([`mr.raps` on CRAN](https://CRAN.R-project.org/package=mr.raps), generated w/ `conda_r_skeleton_helper`)
- `r-mrinstruments` ([`MRInstruments` on R-universe](https://mrcieu.r-universe.dev/MRInstruments))
- `r-mrmix` ([`MRMix` on R-universe](https://mrcieu.r-universe.dev/MRMix))
- `r-mrpresso` ([`MRPRESSO` on R-universe](https://mrcieu.r-universe.dev/MRPRESSO))
- `r-radialmr` ([`RadialMR` on R-universe](https://mrcieu.r-universe.dev/RadialMR))

Apologies in advance for the heavy review! Fortunately all are `noarch`, so only need **linux-64** logs should require reviewing.

## Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

mfansler commented:
@conda-forge/help-r bump

mfansler commented:
@dbast @cbrueffer would either of you have time to review? Thanks 🙏

dbast commented:
@mfansler hm, the packages from https://mrcieu.r-universe.dev/src/contrib/ have some issues:
* not published to CRAN which serves as nice quality gate
* no stable tarball hash: breaks automatic bot rebuilds / migrations
* md5 hash = make un-noticeable supply chain attacks possible, where a system gets hacked + files manipulated while the hash does not change

They seem to have their releases tagged, see e.g. https://github.com/MRCIEU/MRInstruments/tags .. like https://github.com/MRCIEU/MRInstruments/archive/refs/tags/0.3.3.tar.gz 

Switching to those release URLs would solve both the hash issues and github release URLs are also bot compatible (by templating the version in the URL). It seem that this is a large community maintaining things, so I don't mind that those are not on cran.

mfansler commented:
> @mfansler hm, the packages from https://mrcieu.r-universe.dev/src/contrib/ have some issues:
> 
> * not published to CRAN which serves as nice quality gate
> * no stable tarball hash: breaks automatic bot rebuilds / migrations
> * md5 hash = make un-noticeable supply chain attacks possible, where a system gets hacked + files manipulated while the hash does not change
> 
> They seem to have their releases tagged, see e.g. https://github.com/MRCIEU/MRInstruments/tags .. like https://github.com/MRCIEU/MRInstruments/archive/refs/tags/0.3.3.tar.gz
> 
> Switching to those release URLs would solve both the hash issues and github release URLs are also bot compatible (by templating the version in the URL). It seem that this is a large community maintaining things, so I don't mind that those are not on cran.

I've switched the MRCIEU packages (`r-mrinstruments` and `r-twosamplemr`) to GitHub tarballs, but the others lack tags/releases. I suspect one motivation for MRCIEU having a dedicated R-universe channel is to host such non-CRAN dependencies that they require.

I've also changed all the MD5 to SHA256 (manually computed for the R-universe tarballs).

dbast commented:
ah... didn't realise that the other recipes have no tags/release in their repos :/ this gets us some steps further, but bot rebuild / updates are still troublesome :/

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/r-mrmix/meta.yaml```, ```recipes/r-radialmr/meta.yaml```, ```recipes/r-mrpresso/meta.yaml```, ```recipes/r-mr.raps/meta.yaml```, ```recipes/r-twosamplemr/meta.yaml```, ```recipes/r-mrinstruments/meta.yaml```) and found it was in an excellent condition.
